src/romloader.cpp

```diff
@@ -0,0 +1,26 @@
+#include <iostream>
+#include "romloader.hpp"
+
+RomLoader::RomLoader(IFileIO& fio)
+    :fileIO(fio){}
+
+bool RomLoader::load(std::string romName)
+{
+    int fd = fileIO.openFile(romName);
+    if (fd > 0) {
+        uint8_t buff[512];
+        while (int readBit = fileIO.readFile(buff, fd)) {
+            for (int i = 0; i < readBit; i++) {
```

Comment:
you could something better instead of the for + push_back 
`std::copy(buff, buff + readBit, std::back_inserter(_data));`
there is also
`_data.insert(_data.end(), buff, buff + readBit)` 
I think the latest one is more efficiency (since he knows the size it will _data.reserve(). But anyway it doesn't really matter. the ROM file are small.